lib/unison-sqlite/src/Unison/Sqlite/Connection.hs

```diff
@@ -93,34 +93,37 @@ withConnection ::
   (Connection -> m a) ->
   m a
 withConnection name file =
-  bracket (openConnection name file) closeConnection
+  bracket (liftIO (openConnection name file)) (liftIO . closeConnection)
 
 -- Open a connection to a SQLite database.
 openConnection ::
-  MonadIO m =>
   -- Connection name, for debugging.
   String ->
   -- Path to SQLite database file.
   FilePath ->
-  m Connection
+  IO Connection
 openConnection name file = do
-  conn0 <- liftIO (Sqlite.open file)
+  conn0 <- Sqlite.open file `catch` rethrowAsSqliteConnectException name file
   let conn = Connection {conn = conn0, file, name}
-  liftIO (execute_ conn "PRAGMA foreign_keys = ON")
+  execute_ conn "PRAGMA foreign_keys = ON"
   pure conn
 
 -- Close a connection opened with 'openConnection'.
-closeConnection :: MonadIO m => Connection -> m ()
+closeConnection :: Connection -> IO ()
 closeConnection (Connection _ _ conn) =
-  liftIO (Sqlite.close conn)
+  -- FIXME if this throws an exception, it won't be under `SomeSqliteException`
```

Comment:
Did you plan to fix this as part of this PR or do you have plans to come back to this later?

Reply:
No plans to fix it in this PR. I think we could either ignore exceptions from close, or make up a new exception type